### PR TITLE
getMatch works

### PR DESCRIPTION
**Pull request by bavaz1**

### CODE REVIEW

File: sdk/sdk.go [hunk 1]
```diff
@@ -109,7 +116,7 @@ const apiShard = "steam"
 const apiFilter = "players?filter[playerNames]="
 const apiMatches = "matches"
 
-func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Base, error) {
+func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Players, error) {
 	s := []string{apiHost, apiEndpoint, apiShard, apiFilter}
```
Comment (PumpkinSeed) [inline sdk/sdk.go:120]: Use strings.Builder

File: sdk/sdk.go [hunk 2]
```diff
@@ -109,7 +116,7 @@ const apiShard = "steam"
 const apiFilter = "players?filter[playerNames]="
 const apiMatches = "matches"
 
-func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Base, error) {
+func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Players, error) {
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: Use pointer for Players, you can use nil in returns

File: sdk/sdk.go [hunk 3]
```diff
@@ -128,33 +135,33 @@ func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Bas
 
 	r, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		return Base{}, err
+		return Players{}, err
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return nil, err

File: sdk/sdk.go [hunk 4]
```diff
@@ -128,33 +135,33 @@ func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Bas
...
 	}
 	r.Header.Set("Accept", "application/vnd.api+json")
 	r.Header.Set("Authorization", fmt.Sprintf("Bearer %s", apiKey))
 	r = r.WithContext(ctx)
 
 	resp, err := client.Do(r)
 	if err != nil {
-		return Base{}, err
+		return Players{}, err
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return nil, err

File: sdk/sdk.go [hunk 5]
```diff
@@ -128,33 +135,33 @@ func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Bas
...
 	}
 	defer resp.Body.Close()
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return Base{}, err
+		return Players{}, err
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return nil, err

File: sdk/sdk.go [hunk 6]
```diff
@@ -128,33 +135,33 @@ func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Bas
...
 
 	if resp.StatusCode == 200 {
-		var result Base
+		var result Players
 		err = json.Unmarshal(body, &result)
 		return result, err
 	}
 
-	return Base{}, fmt.Errorf("Undefined status code: %s", string(body))
+	return Players{}, fmt.Errorf("Undefined status code: %s", string(body))
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return nil, ....

File: sdk/sdk.go [hunk 7]
```diff
@@ -128,33 +135,33 @@ func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Bas
...
 }
 
-func GetMatch(ctx context.Context, matchID string, client *http.Client) (Base, error) {
+func GetMatch(ctx context.Context, matchID string, client *http.Client) (Match, error) {
 	s := []string{apiHost, apiEndpoint, apiShard, apiMatches, matchID}
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: Use strings.Builder

File: sdk/sdk.go [hunk 8]
```diff
@@ -128,33 +135,33 @@ func GetPlayer(ctx context.Context, playerName string, client *http.Client) (Bas
...
 		err = json.Unmarshal(body, &result)
 		return result, err
 	}
 
-	return Base{}, fmt.Errorf("Undefined status code: %s", string(body))
+	return Players{}, fmt.Errorf("Undefined status code: %s", string(body))
 }
 
-func GetMatch(ctx context.Context, matchID string, client *http.Client) (Base, error) {
+func GetMatch(ctx context.Context, matchID string, client *http.Client) (Match, error) {
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return Match as pointer

File: sdk/sdk.go [hunk 9]
```diff
@@ -173,29 +180,28 @@ func GetMatch(ctx context.Context, matchID string, client *http.Client) (Base, e
 
 	r, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		return Base{}, err
+		return Match{}, err
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return nil, err

File: sdk/sdk.go [hunk 10]
```diff
@@ -173,29 +180,28 @@ func GetMatch(ctx context.Context, matchID string, client *http.Client) (Base, e
...
 	}
 	r.Header.Set("Accept", "application/vnd.api+json")
 	r.Header.Set("Authorization", fmt.Sprintf("Bearer %s", apiKey))
 	r = r.WithContext(ctx)
 
 	resp, err := client.Do(r)
 	if err != nil {
-		return Base{}, err
+		return Match{}, err
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return nil, err

File: sdk/sdk.go [hunk 11]
```diff
@@ -173,29 +180,28 @@ func GetMatch(ctx context.Context, matchID string, client *http.Client) (Base, e
...
 	}
 	defer resp.Body.Close()
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return Base{}, err
+		return Match{}, err
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return nil, err

File: sdk/sdk.go [hunk 12]
```diff
@@ -173,29 +180,28 @@ func GetMatch(ctx context.Context, matchID string, client *http.Client) (Base, e
...
 	if resp.StatusCode == 200 {
-		var result Base
+		var result Match
 		err = json.Unmarshal(body, &result)
-		fmt.Println(result)
 		return result, err
 	}
 
-	return Base{}, fmt.Errorf("Undefined status code: %s", string(body))
+	return Match{}, fmt.Errorf("Undefined status code: %s", string(body))
```
Comment (PumpkinSeed) [inline sdk/sdk.go]: return nil, ...

**Review by PumpkinSeed — CHANGES_REQUESTED** [review]
Write a request method, and it will do the actual request, waits parameters like `method, url, etc.` and returns a []byte of the response and error.

**Comment by bavaz1** [comment]
Request function is added, but i dont get it why should i use strings.Builder. Could you explain that please?

**Review by PumpkinSeed — APPROVED** [review]

**Pull request merged**